### ISSUE TITLE
IDISP006 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#192]
```cs
namespace N
{
    using System;
    using System.IO;

    abstract class BaseClass : IDisposable
    {
        public void Dispose()
        {
            this.M();
        }

        protected abstract void M();
    }

    class C : BaseClass
    {
        private readonly Stream stream = File.OpenRead(string.Empty);

        protected override void M()
        {
            this.stream.Dispose();
        }
    }
}
```

Also IDISP002 can do a better job figuring it out.
